### ISSUE TITLE
6502 emulation imprecision: opcode 6c: JMP indirect

### ISSUE THREAD

**Issue by Shachar** [AppleWin/AppleWin#989]
The original 6502, when asked to jump indirect, and the address to jump to was stored on the last byte of the page (e.g. $2ff), would load the address to jump to from that address and the first byte **of the same page**, rather than the first byte of the next one.
Try the following sequence of commands:
```
] CALL -151
* 0310:6c ff 02 // JMP ($2ff)
* 2ff:20 03 // Make 2ff, 300 point to 0320
* 200:08   // Make 2ff, 200 point to 0820
* 0820:00 // BRK
* 0320:00 // BRK
* 310G
```
Under the 65c02 the BRK will show PC of 322, which means it jumped to the BRK at $320, as it should. Under the original 6502, however, it should have jumped to $0820. In AppleWin 6502 emulation, it behaves like the 65c02.

**Comment by tomcw** [comment]
tl;dr - this bug report is invalid - ie. AppleWin's NMOS 6502 implementation of JMP (INDIRECT) is working correctly.

Your repro is broken, as $200 is used to store the keyboard input... so your `200:08` gets overwritten by the subsequent things you type, concluding with the `310G`, which sets $200 to high-bit Ascii "3" or $B3. So the code when run on a 6502 actually JMPs to $B320, which then depends on what is at that memory location. For me I get a monitor break at $B325. 

Try this:  (just copy to the Windows clipboard, then SHIFT+INSERT to paste into AppleWin)
310:6C FF 20
20FF:20 03
2000:20
2020:0
320:0
310G

The result on an NMOS 6502, is a monitor break at $2022.
And on a CMOS 65C02, is a monitor break at $322.

**Comment by tomcw** [comment]
btw, there is a minor disassembly bug for the NMOS 6502, where the built-in debugger shows the JMP (INDIRECT) opcode's target address as $320...

<img width="309" alt="debugger-bug" src="https://user-images.githubusercontent.com/6696896/136851268-e75bdc67-be72-443f-a7cf-08d3f70e7f7e.png">

**Comment by tomcw** [comment]
Fixed this debugger NMOS target address issue in 028c041.
